src/components/PopupWithForm.js

```diff
@@ -27,10 +28,38 @@ class PopupWithForm extends Popup {
     super.setEventListeners();
     this._form.addEventListener('submit', (e) => {
       e.preventDefault();
-      this._handleFormSubmit(this._getInputValues());
-      this._form.reset(); // Clear inputs after form submission
+      this.renderLoading(true);
+
+      const submitResult = this._handleFormSubmit(this._getInputValues());
+      
+      if (submitResult && typeof submitResult.then === 'function') {
+        submitResult
+          .then(() => {
+            this._form.reset(); // Clear inputs after successful form submission
+            this._submitButton.disabled = true; // Disable the submit button
+            this._submitButton.classList.add('modal__button_disabled');
```

Comment:
Now only class `FormValidator` should **`disable/enable`** submit buttons and remove errors from the inputs (if you do it). It’s a violation of **Single responsibility** principle